index.bs

```diff
@@ -237,7 +334,28 @@ To <dfn>obtain an attribution source</dfn> from an <{a}> element |anchor|:
 <dfn>Max event id value</dfn> is a vendor specific integer which controls 
 the maximum size value which can be used as an [=attribution source/event id=]
 
-Issue: Need to spec how to store the attribution source.
+<h3 id="processing-an-attribution-source">Processing an attribution source</h3>
+
+To <dfn>maybe process an attribution source</dfn> given a <a spec="HTML">navigation params</a>
+|navigationParams| and [=browsing context=] |browsingContext|, run the following steps:
+1. If |browsingContext| is not a <a spec="html">top-level browsing context</a>, return.
+1. Let <var>attributionSource</var> be |navigationParams|'s <a lt="navigation params attribution source">attribution source</a>.
+1. If |attributionSource| is null, return.
+1. Let |attributionSource|'s [=attribution source/attribution destination=] is not [=same site=] to |navigationParams|'s 
+    <a href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigation-params-origin">origin</a>, return.
+1. [=Queue a task=] to [=process an attribution source=] with |attributionSource|.
+
+To <dfn>process an attribution source</dfn> given an [=attribution source=] |source|:
+
+1. Let |cache| be the user agent's [=attribution source cache=].
+1. Let |sources| be the all entries in |cache| where:
+    1. the entry's [=attribution source/attribution destination=] matches |source|'s [=attribution source/attribution destination=].
+    1. the entry's [=attribution source/reporting endpoint=] matches |source|'s [=attribution source/reporting endpoint=].
+    1. the entry's [=attribution source/num reports=] value is greater than 0 .
+1. Remove |sources| from |cache|.
```

Comment:
Since you don't use |sources| later, I'd use something like:

```suggestion
1. [=list/Remove=] all entries in |cache| where any of:
    1. the entry's [=attribution source/attribution destination=] matches |source|'s [=attribution source/attribution destination=].
    1. the entry's [=attribution source/reporting endpoint=] matches |source|'s [=attribution source/reporting endpoint=].
    1. the entry's [=attribution source/num reports=] value is greater than 0 .
```